storage/innobase/include/os0enc.h

```diff
@@ -32,11 +32,19 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301  USA
 
 #include "keyring_encryption_key_info.h"
 #include "template_utils.h"
+#include "page0types.h"
 
 #include "univ.i"
 
 // Forward declaration.
 class IORequest;
+struct fil_space_t;
+namespace file {
+struct Block;
+}
```

Comment:
I would also declare
```c++
struct block_deleter {
  void operator() (Block* ptr) const noexcept { os_free_block(ptr); }
};
using block_ptr = std::unique_ptr<Block, block_deleter>;
```
here that can be used as an output parameter for `Double_write::prepare()`, `Encryption::dblwr_encrypt_page()` and `Encryption::dblwr_decrypt_page()`